server/services/service.js

```diff
@@ -74,16 +74,21 @@ module.exports = () => ({
       if (config.drafts) return true;
       return x.publishedAt;
     });
-
-    return dataFiltered.map((x) => ({
-      id: x.id,
-      title: config.titleField ? x[config.titleField] : config.startField,
-      startDate: x[config.startField],
-      endDate: config.endField
-        ? x[config.endField]
-        : moment(x[config.startField]).add(config.defaultDuration, 'minutes'),
-      color: config.colorField ? x[config.colorField] : null,
-    }));
+    return dataFiltered.map((x) => {
+      const startDate = x[config.startField].split('T')[0];
+      const endDate = config.endField ? x[config.endField].split('T')[0] : null; 
+      const isMultiDay = endDate && moment(endDate).isAfter(startDate); 
+      const startDateTime = `${startDate}T00:00:00`; 
+      const endDateTime = isMultiDay ? `${endDate}T00:00:00` : moment(startDateTime).add(config.defaultDuration, 'minutes').format(); 
+      return {
+         id: x.id,
+         title: config.titleField ? x[config.titleField] : x[config.startField],
+         startDate: startDateTime,
+         endDate: endDateTime,
+         color: config.colorField ? x[config.colorField] : null,
+         allDay: isMultiDay, 
```

Comment:
I’m a bit confused by the request to remove the allDay attribute. The primary aim of this pull request was to ensure that full-day and multi-day events are accurately represented in the calendar, and this relies on the allDay attribute. Without it, handling multi-day events, particularly across time zones, becomes more challenging.

Since the allDay attribute will be required in the upcoming version, wouldn’t it make sense to retain it in this PR for future compatibility and to ensure proper rendering now? I’d appreciate it if you could clarify the reasoning behind its removal, given its role in managing these event types.

Thanks

Reply:
Apologies.
Upon further inspection it turns out the allDay attribute **is** needed (Demo [here](https://codesandbox.io/p/sandbox/lvhdrt?file=%2Fdemo-data%2Fappointments.js))